grpc/src/main/java/com/linecorp/armeria/internal/server/grpc/AbstractServerCall.java

```diff
@@ -537,11 +539,20 @@ protected final HttpObject responseTrailers(ServiceRequestContext ctx, Status st
     // Returns ResponseHeaders if headersSent == false or HttpHeaders otherwise.
     public static HttpHeaders statusToTrailers(
             ServiceRequestContext ctx, HttpHeadersBuilder trailersBuilder, Status status, Metadata metadata) {
+        try {
+            MetadataUtil.fillHeaders(metadata, trailersBuilder);
+        } catch (Exception e) {
+            // Continue by squashing Exception by catch,
+            // because server implementer could have set corrupted metadata.
+            logger.warn("{} {} status: {}", ctx, "Error at filling header with metadata.", status);
+            // Catching exception is necessary, because server implementer could have set corrupted metadata.
```

Comment:
We could clean up the comments and logging sentence a little bit:
```suggestion
            // A buggy user-implemented custom metadata serializer may throw
            // an exception. Leave a log message and set the INTERNAL status.
            logger.warn("{} Failed to serialize metadata; overriding the original status ({}) with INTERNAL:", ctx, status, e);
```